Fixing Dial-graphs misaligned in Safari/OS X

simonnagl opened the pull request:
This fixes the misalignment described in #159.

I have tested the pull request on
* Safari 9.1 (OS X)
* Chromium 50.0.2661.75 and (Linux)
* Mozilla Firefox 45.02. (Linux )

I made this fix in two commits because I do not know if we want to support old browsers.
I also want to admit that this might destroy layout for very old browsers.

Fore more information look at [w3schools](http://www.w3schools.com/css/css3_flexbox.asp)

ktsaou commented:
Have you tested this on firefox, chrome, etc? Do you want me to do it before merging it?
This repo is cloned more than 700 times per day, so I try to be careful...

I have an account on browserstack.com, so if you have it public somewhere I can check it on different browser versions.

simonnagl commented:
As described above I tested it with

* Safari 9.1 (OS X)
* Chromium 50.0.2661.75 and (Linux)
* Mozilla Firefox 45.02. (Linux )

Testing in my case means looking and scrolling the site in the described browsers with different window sizes and their "Responsive Design Modes"

I total understand that you are careful. Testing this changes with browserstack.com seems a good idea.

I am running netdata at home and do not want to make it public. Maby someone else of the community want's to check out my changes and make the site public for testing.
I also have no browserstack.com account. Maybe you can test it on the demo site. For testing it you only need to change two files and the worst thing which could happen is misalignment of some content.

Sorry that I cannot test this carefully.

ktsaou commented:
ok, there is something with this I don't like.

This is the old (available at http://netdata.firehol.org?nowelcome):

![old1](https://cloud.githubusercontent.com/assets/2662304/14800033/fd23a522-0b47-11e6-8754-6f480671aaeb.gif)

Notice that the page comes at a fixed position.

Now this PR (available at http://netdata2.firehol.org?nowelcome):

![new1](https://cloud.githubusercontent.com/assets/2662304/14800059/2389a900-0b48-11e6-8665-916e7bd3b600.gif)

The page jumps while rendering the gauges at the top (the gif does not show the full detail).

In general, I try to have the page come fixed from beginning. Because charts are created and destroyed all the time. To make it appear, try this:

Go to to settings and switch "Hide" to "Destroy". Now scroll the page to move off gauges off the screen and come back again. The whole page jumps...

ktsaou commented:
ok. the primary demo site died and I switched the servers.
You cannot test it live now.

simonnagl commented:
Thank you for testing! I totally see your point. I did not test the site with the configuration destroy. A jumping site when scrolling is not what we want.

I will debug and understand what "destroy" does technically in detail and make a suggestion how to fix this until the beginning of the next week.

ktsaou commented:
Perfect!

ktsaou commented:
I will add the `wontmerge` tag until you push an update. Don't get disappointed by it. It is just a flag for me to remember I should not merge this until something happens.

ktsaou commented:
of course, post something here when you make a new commit, so I will be notified.

simonnagl commented:
I rebased to master and added a new commit.

### Description
**Before the pull request**
The elements of the row where placed as inline-blocks into a centerd div with style block.
This is "right" but there is a rendering issue with Safari.
**Before the last commit**
The elements of the row where placed in a flex-items into a flex-box div which had the property align-items: baseline.
This is "right" but there is a rendering issue with Safari.
**Now**
The elements of the row where placed in a flex-items into a flex-box div which had the property align-items: flex-end.
In this context flex-end has the same meaning like baseline.
This is "right" and there is hopefully no rendring issue with any browser.

Right means that all three variants mean the same layout seen by the user.

@ktsaou Could you please retest this? Thank you!

ktsaou commented:
of course! But give me a day. I need to travel tonight.

ktsaou commented:
It took a little longer. I will setup a demo now to test it.

ktsaou commented:
Excellent job so far. I am still testing it.

The only problem I found was this on IOS:

![image](https://cloud.githubusercontent.com/assets/2662304/14995256/8817e0fa-117c-11e6-8bd0-608d1e7c7f8b.png)

I tested also on my real iphone. It happens occasionally (it happens on upstream/master too).

ktsaou commented:
On safari 8.0 too:

![image](https://cloud.githubusercontent.com/assets/2662304/14995400/405b651a-117d-11e6-8c2a-4a607245e65f.png)

ktsaou commented:
And safari 7.1 the same. It is strange because after a while it goes away. You have to refresh the page to see it again.

Since this does not appear in safari 9 (neither chrome, firefox, opera), I say to merge it.
You have done a good job.

Do you agree?

ktsaou commented:
You can test your PR at: http://46.101.213.117:19999/

simonnagl commented:
Yes, I agree. Thank you for testing this!

On w3schools I found [that table](http://www.w3schools.com/cssref/css3_browsersupport.asp). It tells you that flex is introduced with version 9. Browsers wich do not support flex fall back to the previous layout. The rendering problem might be the same as before in all Safari versions.

ktsaou commented:
merged!
